### PR TITLE
GOB: Switch detection entries from GOB3 & onceupon to AD_ENTRY & add english version of Adibou 3 to detection

### PR DESCRIPTION
**Pull request by BJNFNE**

### CODE REVIEW

**Comment by lephilousophe** [comment]
This is a lot of PR to change detection entries.
As soon as one is closed, another is opened.
Would it be possible to make a single PR with all the changes you planned?

In addition there are several alignment problems in this PR.
Please use tabs for indentation but spaces for alignment.

Thanks.

**Comment by BJNFNE** [comment]
> This is a lot of PR to change detection entries. As soon as one is closed, another is opened. Would it be possible to make a single PR with all the changes you planned?
> 
> In addition there are several alignment problems in this PR. Please use tabs for indentation but spaces for alignment.
> 
> Thanks.

Thanks for the Reply, in VSCode where i did the Changes everything looks correct.

> This is a lot of PR to change detection entries. As soon as one is closed, another is opened. Would it be possible to make a single PR with all the changes you planned?
> 
> In addition there are several alignment problems in this PR. Please use tabs for indentation but spaces for alignment.
> 
> Thanks.

Thanks for the message.
i have fixed the problem with the alignments for onceupon

**Comment by lephilousophe** [comment]
I still see misalignment between hashes and file names in the Github preview.

**Comment by BJNFNE** [comment]
only for onceupon or aswell in GOB3?

File: engines/gob/detection/tables_gob3.h [hunk 1]
```diff
@@ -194,11 +194,8 @@
 	{
 		"gob3",
 		"v1.00",
-		{
-			{"intro.stk", 0, "16b014bf32dbd6ab4c5163c44f56fed1", 445104},
-			{"musmac1.mid", 0, "948c546cad3a9de5bff3fe4107c82bf1", 6404},
-			{0, 0, 0, 0}
-		},
+		AD_ENTRY2s("intro.stk", "16b014bf32dbd6ab4c5163c44f56fed1", 445104,
+				"musmac1.mid",	"948c546cad3a9de5bff3fe4107c82bf1", 6404),
```
Comment (lephilousophe) [inline engines/gob/detection/tables_gob3.h:198]:
```suggestion
		AD_ENTRY2s("intro.stk",   "16b014bf32dbd6ab4c5163c44f56fed1", 445104,
		           "musmac1.mid", "948c546cad3a9de5bff3fe4107c82bf1", 6404),
```

File: engines/gob/detection/tables_onceupon.h [hunk 2]
```diff
@@ -36,12 +36,9 @@
...
 		"",
-		{
-			{"stk1.stk", 0, "a8e963eea170155548e5bc1d0f07d50d", 209806},
-			{"stk2.stk", 0, "e4b21818af03930dc9cab2ad4c93cb5b", 362106},
-			{"stk3.stk", 0, "76874ad92782f9b2de57beafc05ec877", 353482},
-			{0, 0, 0, 0}
-		},
+		AD_ENTRY3s("stk1.stk", "a8e963eea170155548e5bc1d0f07d50d", 209806,
+			"stk2.stk",	"e4b21818af03930dc9cab2ad4c93cb5b", 362106,
+			"stk3.stk",	"76874ad92782f9b2de57beafc05ec877", 353482),
```
Comment (lephilousophe) [inline engines/gob/detection/tables_onceupon.h:41]:
```suggestion
		AD_ENTRY3s("stk1.stk", "a8e963eea170155548e5bc1d0f07d50d", 209806,
		           "stk2.stk", "e4b21818af03930dc9cab2ad4c93cb5b", 362106,
		           "stk3.stk", "76874ad92782f9b2de57beafc05ec877", 353482),
```

**Review by lephilousophe — COMMENTED** [review]
Both of them.

Please note how I removed all tabs you inserted except for the first ones used to indent the code and replaced them by spaces at appropriate places to get file names, hashes and file sizes starting at the same column for all the rows in the entry.

Old gob3 entries were not really aligned but all the curly braces started at the same column.
On old onceupon, all entries having the same name length were all aligned.

**Comment by BJNFNE** [comment]
Thank you, i applied your suggested changes

**Comment by lephilousophe** [comment]
What I suggested was supposed to be replicated to every entries in both files.

**Comment by lephilousophe** [comment]
I had to fix the alignment myself.

**Comment by lephilousophe** [comment]
It will when you will add the new detection entry you will have soon

**Comment by larsamannen** [comment]
> whats now the current status of the PR can this be merged now or not yet?

We really appreciate your passion to add meta data to the detection entries for the GOB engine.
What we ask of you is that instead of creating tiny PRs with only one or two commits, adding or changing version numbers, indentations or other things which is not really needed for the detection to work, we ask you to do what you do but collect your work over longer times so the PRs contains a bit more.
Every time a PR is opened this requires someone from the team to review and validate the changes. This is quite time consuming so I think everyone would gain of creating bigger PRs but less often. That way you'll also get some time to verify that small mistakes aren't sneaked in unintentionally and the review can be done hopefully only once.

So the message is; keep doing what you're doing, we depend on passionate people caring for the project, but extend your work over longer periods so the detection entries becomes more maintainable.

Cheers

**Comment by BJNFNE** [comment]
> > whats now the current status of the PR can this be merged now or not yet?
> 
> 
> 
> We really appreciate your passion to add meta data to the detection entries for the GOB engine.
> 
> What we ask of you is that instead of creating tiny PRs with only one or two commits, adding or changing version numbers, indentations or other things which is not really needed for the detection to work, we ask you to do what you do but collect your work over longer times so the PRs contains a bit more.
> 
> Every time a PR is opened this requires someone from the team to review and validate the changes. This is quite time consuming so I think everyone would gain of creating bigger PRs but less often. That way you'll also get some time to verify that small mistakes aren't sneaked in unintentionally and the review can be done hopefully only once.
> 
> 
> 
> So the message is; keep doing what you're doing, we depend on passionate people caring for the project, but extend your work over longer periods so the detection entries becomes more maintainable.
> 
> 
> 
> Cheers

Thank you, I will take care of that.

**Comment by BJNFNE** [comment]
Regarding @bluegr PR about GOB and the new gametypetoid change, have i rebase the PR?

**Comment by sev-** [comment]
Now this contains a merge commit, but it must be rebased.

**Comment by BJNFNE** [comment]
This PR is now ready to be merged

**Comment by lephilousophe** [comment]
This has been over a month this PR is opened and you continue to add things to tables (which is great!).
Do you see a point in time where it will settle?
Should we merge or can you squeeze other changes in it?

**Comment by BJNFNE** [comment]
This can be merged, i have currently nothing ToDo on GOB

**Comment by lephilousophe** [comment]
Thanks! I merge it now.

**Pull request merged**